ignite/pkg/cosmosanalysis/app/app.go

```diff
@@ -99,24 +102,45 @@ func FindRegisteredModules(chainRoot string) ([]string, error) {
 		return nil, err
 	}
 
-	var basicModules []string
+	// The directory where the app file is located.
+	// This is required to resolve references within the app package.
+	appDir := filepath.Dir(appFilePath)
+
 	ast.Inspect(f, func(n ast.Node) bool {
-		if pkgsReg := findBasicManagerRegistrations(n, basicManagerModule); pkgsReg != nil {
-			for _, rp := range pkgsReg {
-				importModule := packages[rp]
-				basicModules = append(basicModules, importModule)
+		// Stop traversing the child nodes when there is an error
+		if err != nil {
+			return false
+		}
```

Comment:
Will this statement ever be hit?  Inside this block, it looks like we are checking `err` whenever it is immediately assigned to already.

Reply:
If I understand the question, the `err` is checked there because the `ast.Inspect` function would, depending on the case, continue parsing the rest of child nodes after an error in the basic manager discovery process was found. According to the `ast.Inspect` documentation:

```
...If f returns true, Inspect invokes f recursively for each of the non-nil children of node, ...
```

Which means that during the children traversing it will continue if we find an error parsing any of the children. So the intention of the statement at the beginning is to stop the traversing to avoid parsing the rest of them.

Reply:
This statement is one way to add error support during inspection.

@tbruyelle proposed another way that will be added later which is to implement a custom `pkg/xast` package and a function that would implement AST inspection with error support.